src/main/kotlin/no/nav/syfo/person/skjermingskode/SkjermingskodeService.kt

```diff
@@ -1,32 +1,56 @@
 package no.nav.syfo.person.skjermingskode
 
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
 import no.nav.syfo.client.pdl.PdlClient
 import no.nav.syfo.client.skjermedepersonerpip.SkjermedePersonerPipClient
 import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.person.api.domain.PersonInfo
 import no.nav.syfo.person.api.domain.Skjermingskode
 
 class SkjermingskodeService(
     private val skjermedePersonerPipClient: SkjermedePersonerPipClient,
     private val pdlClient: PdlClient,
 ) {
-    suspend fun hentBrukersSkjermingskode(
+    suspend fun hentSkjermingskodeForPersonidenter(
         callId: String,
-        personIdent: PersonIdentNumber,
+        personidenter: List<PersonIdentNumber>,
         token: String,
-    ): Skjermingskode? {
-        val hasAdressebeskyttelse = pdlClient.hasAdressebeskyttelse(callId = callId, personIdent = personIdent)
+    ) =
+        personidenter.map { personident ->
+            val skjermingskode = hentBrukersSkjermingskode(
+                callId = callId,
+                personident = personident,
+                token = token,
+            )
+            Pair(personident, skjermingskode)
+        }.map { (personident, skjermingskode) ->
+            PersonInfo(
+                fnr = personident.value,
+                skjermingskode = skjermingskode.await(),
+            )
+        }
 
-        return hasAdressebeskyttelse?.let {
-            when {
-                hasAdressebeskyttelse -> Skjermingskode.DISKRESJONSMERKET
-                else -> if (
-                    skjermedePersonerPipClient.isSkjermet(
+    private suspend fun hentBrukersSkjermingskode(
+        callId: String,
+        personident: PersonIdentNumber,
+        token: String,
+    ): Deferred<Skjermingskode> =
+        CoroutineScope(Dispatchers.IO).async {
+            val hasAdressebeskyttelse = pdlClient.hasAdressebeskyttelse(callId = callId, personIdent = personident)
+            hasAdressebeskyttelse?.let {
+                if (hasAdressebeskyttelse) {
+                    Skjermingskode.DISKRESJONSMERKET
+                } else {
+                    val isSkjermet = skjermedePersonerPipClient.isSkjermet(
                         callId = callId,
-                        personIdentNumber = personIdent,
+                        personIdentNumber = personident,
                         token = token,
                     )
-                ) Skjermingskode.EGEN_ANSATT else Skjermingskode.INGEN
-            }
+                    if (isSkjermet) Skjermingskode.EGEN_ANSATT else Skjermingskode.INGEN
+                }
+            } ?: Skjermingskode.INGEN
```

Comment:
Er dette det samme som å returnere `null` for der `Skjermingskode?` ikke hadde verdi? 🤔

Reply:
Testene forteller meg at det har noe å si 🤌